src/Twig/RolesMatrixExtension.php

```diff
@@ -71,14 +71,15 @@ public function renderMatrix(Environment $environment, FormView $form): string
     {
         $groupedRoles = [];
         foreach ($this->rolesBuilder->getRoles() as $role => $attributes) {
-            if (!isset($attributes['admin_label'])) {
+            if (!isset($attributes['admin_code'])) {
```

Comment:
This is a BC-break for people using this extensions with their own data.

If we change that much the job of this method
- either we create a new one
- either we keep BC with deprecation for the old behavior.

Reply:
Attribute "admin_label" can repeat not even in different groups but also in the same group. I don't know how to avoid this change. If group not set or "show_in_dashboard" set to "false" admin does not have group.

Reply:
Yeah I understand. That's maybe why the previous suggestion was to use `Group - Name` in the same value, rather than having a group key and a label key.

What you may do is 
```
if (!isset($attributes['admin_code'])) {
    // NEXT_MAJOR: Remove this code.
    if (isset($attributes['admin_label']) {
        // trigger a deprecation

        // apply the old behavior
    }

     continue;
```
WDYT, would it be possible to do something like this ?

Reply:
> Yeah I understand. That's maybe why the previous suggestion was to use `Group - Name` in the same value, rather than having a group key and a label key.

To avoid repeat labels need something like:
```
$groupLabels=[];
$adminLabels=[];

$i=0; $k=0;
foreach ($admins as $admin){
    if(!isset($groupLabels[$admin['group_code']))
    {
        $groupLabel=$admin['group_label'];
        if(array_search($admin['group_label'], $groupLabels, true)!==false)
        {
            $groupLabel=$admin['group_label'].'('.++$i.')';
        }

        $groupLabels[$admin['group_code']]=$groupLabel;
    }

    if(!isset($adminLabels[$admin['group_code']][$admin['admin_code']))
    {
        $adminLabel=$admin['admin_label'];
        if(array_search($admin['admin_label'], $adminLabels[$admin['group_code']], true)!==false)
        {
            $adminLabel=$admin['admin_label'].'('.++$k.')';
        }

        $adminLabels[$admin['group_code']][$admin['admin_code']]=$adminLabel;
    }
}
```
Result 
Group_1 - Name_1,
Group_1(1) - Name_2,
Group_2 - Name_1,
Group_2(2) - Name_1(1),

> WDYT, would it be possible to do something like this ?
```
if (!isset($attributes['admin_code'])) {
    // NEXT_MAJOR: Uncomment
    // continue;

    // NEXT_MAJOR: Remove this code.
    // trigger a deprecation
    if (!isset($attributes['admin_label'])) {
        continue;
    }

    $attributes['admin_code'] = $attributes['admin_label'];
}
```
How to properly trigger a deprecation?

It is better implement deprecation or Group - Name ?

Reply:
> How to properly trigger a deprecation?

We're using
```
        @trigger_error(sprintf(
            'Foo is deprecated since sonata-project/foo-lib 42.x and will be removed in version 43.0.',
        ), \E_USER_DEPRECATED);
```

> It is better implement deprecation or Group - Name ?

I'm fine with both. I just wanted to avoid the BC-break.